ambari-server/src/main/java/org/apache/ambari/server/checks/AmbariMetricsHadoopSinkVersionCompatibilityCheck.java

```diff
@@ -0,0 +1,215 @@
+    return Sets.newHashSet("AMBARI_METRICS");
+  }
+
+  @Override
+  public void perform(PrerequisiteCheck prerequisiteCheck, PrereqCheckRequest prereqCheckRequest) throws AmbariException {
+
+    AmbariManagementController ambariManagementController = AmbariServer.getController();
+
+    ResourceProvider provider = AbstractControllerResourceProvider.getResourceProvider(
+      Resource.Type.Request,
+      ambariManagementController
+      );
+
+    String clusterName = prereqCheckRequest.getClusterName();
+
+    Set<String> hosts = ambariManagementController.getClusters()
+      .getCluster(clusterName).getHosts("AMBARI_METRICS", "METRICS_MONITOR");
+
+    Set<Map<String, Object>> propertiesSet = new HashSet<>();
+    Map<String, Object> properties = new LinkedHashMap<>();
+    properties.put(RequestResourceProvider.REQUEST_CLUSTER_NAME_PROPERTY_ID, clusterName);
+
+    Set<Map<String, Object>> filterSet = new HashSet<>();
+    Map<String, Object> filterMap = new HashMap<>();
+    filterMap.put(RequestResourceProvider.SERVICE_ID, "AMBARI_METRICS");
+    filterMap.put(RequestResourceProvider.COMPONENT_ID, "METRICS_MONITOR");
+    filterMap.put(RequestResourceProvider.HOSTS_ID, StringUtils.join(hosts,","));
+    filterSet.add(filterMap);
+
+    properties.put(RequestResourceProvider.REQUEST_RESOURCE_FILTER_ID, filterSet);
+    propertiesSet.add(properties);
+
+    Map<String, String> requestInfoProperties = new HashMap<>();
+    requestInfoProperties.put(RequestResourceProvider.COMMAND_ID, "CHECK_HADOOP_SINK_VERSION");
+    requestInfoProperties.put(RequestResourceProvider.REQUEST_CONTEXT_ID, "Pre Upgrade check for compatible Hadoop Metric " +
+      "Sink version on all hosts.");
+
+    Request request = PropertyHelper.getCreateRequest(propertiesSet, requestInfoProperties);
+    try {
+      org.apache.ambari.server.controller.spi.RequestStatus response = provider.createResources(request);
+      Resource responseResource = response.getRequestResource();
+      String requestIdProp = PropertyHelper.getPropertyId("Requests", "id");
+      long requestId = (long) responseResource.getPropertyValue(requestIdProp);
+      LOG.debug("RequestId for AMS Hadoop Sink version compatibility pre check : " + requestId);
+
+      Thread.sleep(RETRY_INTERVAL);
+      PreUpgradeCheckStatus status;
+      int retry = 0;
+      LinkedHashSet<String> failedHosts = new LinkedHashSet<>();
+      while ((status = pollRequestStatus(requestId, failedHosts)).equals(PreUpgradeCheckStatus.RUNNING)
+        && retry++ < NUM_TRIES) {
+        if (retry != NUM_TRIES) {
+          Thread.sleep(RETRY_INTERVAL);
+        }
+      }
+
+      if (status.equals(PreUpgradeCheckStatus.SUCCESS)) {
+        prerequisiteCheck.setStatus(PrereqCheckStatus.PASS);
+      } else {
+        prerequisiteCheck.setStatus(PrereqCheckStatus.FAIL);
+        prerequisiteCheck.setFailReason(FAILED_REASON);
+        prerequisiteCheck.setFailedOn(failedHosts);
```

Comment:
This check should be defined as a HOST one since you're using `failedOn` with hosts.  The CheckDescription as SERVICE should be a list of service names for `failedOn`

Reply:
Will do.